coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,680 @@
+            raise ValueError("No `output_regex` specified.")
+
+        options["output_regex"] = re.compile(options["output_regex"])
+
+        # Don't setup severity_map if one is provided by user or if it's not
+        # used inside the output_regex. If one is manually provided but not
+        # used in the output_regex, throw an exception.
+        if "severity_map" in options:
+            if "severity" not in options["output_regex"].groupindex:
+                raise ValueError("Provided `severity_map` but named group "
+                                 "`severity` is not used in `output_regex`.")
+            assert_right_type(options["severity_map"], dict, "severity_map")
+
+            for key, value in options["severity_map"].items():
+                try:
+                    assert_right_type(key, str, "<severity_map dict-key>")
+                except TypeError:
+                    raise TypeError("The key " + repr(key) + " inside given "
+                                    "severity-map is no string.")
+
+                try:
+                    assert_right_type(value, int, "<severity_map dict-value>")
+                except TypeError:
+                    raise TypeError(
+                        "The value {} for key {} inside given severity-map is "
+                        "no valid severity value.".format(repr(value),
+                                                          repr(key)))
+
+                if value not in RESULT_SEVERITY.reverse:
+                    raise TypeError(
+                        "Invalid severity value {} for key {} inside given "
+                        "severity-map.".format(repr(value), repr(key)))
+
+            # Auto-convert keys to lower-case. This creates automatically a new
+            # dict which prevents runtime-modifications.
+            options["severity_map"] = {
+                key.lower(): value
+                for key, value in options["severity_map"].items()}
+
+        allowed_options |= {"output_regex", "severity_map"}
+    elif options["output_format"] is not None:
+        raise ValueError("Invalid `output_format` specified.")
+
+    if options["prerequisite_check_command"]:
+        if "prerequisite_check_fail_message" in options:
+            assert_right_type(options["prerequisite_check_fail_message"],
+                              str,
+                              "prerequisite_check_fail_message")
+        else:
+            options["prerequisite_check_fail_message"] = (
+                "Prerequisite check failed.")
+
+        allowed_options.add("prerequisite_check_fail_message")
+
+    # Check for illegal superfluous options.
+    superfluous_options = options.keys() - allowed_options
+    if superfluous_options:
+        raise ValueError(
+            "Invalid keyword arguments provided: " +
+            ", ".join(repr(s) for s in sorted(superfluous_options)))
+
+
+@enforce_signature
+def Linter(executable: str,
```

Comment:
actually I agree, a decorator is always a function and should start lowercase. How much work is it to change that? I assume like 10 mins because of all the commits on coala-bears?